substrate/frame/support/src/lib.rs

```diff
@@ -2262,6 +2262,221 @@ pub mod pallet_macros {
+	/// externally.
+	///
+	/// Other than the `fn` attached to `Pallet`, this block will also generate an `enum Call`
+	/// which encapsulates the different functions, alongside their arguments, except for `origin`.
+	/// [`sp_runtime::traits::Dispatchable`] is then implemented for `enum Call`.
+	///
+	/// ## Example:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     use frame_support::pallet_prelude::*;
+	///
+	///     #[pallet::config]
+	///     pub trait Config: frame_system::Config { }
+	///
+	///     #[pallet::pallet]
+	///     pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::call]
+	/// 	impl<T: Config> Pallet<T> {
+	/// 		fn do_stuff(_origin: OriginFor<T>, arg: u32) {
+	/// 			unimplemented!()
+	/// 		}
+	/// 	}
+	/// }
+	/// ```
+	///
+	/// TODO: Once we have default pallet config, it would also be easy to create types in the example
+	/// that implement `Config`.
+	pub use frame_support_procedural::call;
+
+	/// Declares a type alias as a storage item. Storage items are pointers to data stored on-chain (the
+	/// *blockchain state*), under a specific key. The exact key is dependent on the type of the
+	/// storage.
+	/// 
+	/// > Hypothetically, one can directly manipulate the state via [`sp_io::storage`](sp_io::storage). However, this is
+	/// > an advance usage and is not recommended.
+	/// 
+	/// ## Storage Types
+	/// 
+	/// The following storage types are supported by the FRAME macros.
+	///
+	/// * [`StorageValue`](frame_support::storage::types::StorageValue)
+	/// * [`StorageMap`](frame_support::storage::types::StorageMap)
+	/// * [`CountedStorageMap`](frame_support::storage::types::CountedStorageMap)
+	/// * [`StorageDoubleMap`](frame_support::storage::types::StorageDoubleMap)
+	/// * [`StorageNMap`](frame_support::storage::types::StorageNMap)
+	/// * [`CountedStorageNMap`](frame_support::storage::types::CountedStorageNMap)
+	///
+	/// The FRAME macros always generate a type alias for one of these types, as indicated by the syntax,
+	/// for example: `type Foo = StorageValue<..>`. For specific information about each storage type, refer to
+	/// the documentation of the respective type.
+	///
+	/// ### Example:
+	/// 
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     # use frame_support::pallet_prelude::*;
+	///     # #[pallet::config]
+	///     # pub trait Config: frame_system::Config {}
+	///     # #[pallet::pallet]
+	///     # pub struct Pallet<T>(_);
+	/// 	#[pallet::storage]
+	/// 	type Foo<T> = StorageValue<_, u32>;
+	///
+	/// 	#[pallet::storage]
+	/// 	type Bar<T> = StorageMap<_, Blake2_128Concat, u32, u32>;
+	///
+	/// 	#[pallet::storage]
+	/// 	type Baz<T> = CountedStorageMap<_, Blake2_128Concat, u32, u32>;
+	/// }
+	/// ```
+	/// 
+	/// ## Related Macros
+	/// 
+	/// The following macros can be used in conjunction with the storage macro:
+	///
+	/// * [`macro@getter`]: Creates a custom getter function.
```

Comment:
I think these don't fit great into the story so far. Moving them toward the end might improve, but no sure. Do you think they are well explained in this position, in the overall narrative?